internal/bsr/container.go

```diff
@@ -28,11 +28,39 @@ const (
 	checksumFileName        = "SHA256SUM"
 	sigFileName             = "SHA256SUM.sig"
 	journalFileName         = ".journal"
+	bsrFailedCheckTemplate  = "FAILED: %s"
+)
+
+// Valid BSR checksum status values.
+const (
+	BSRCheckPass BSRChecksumStatus = "OK"
 )
 
 // ContainerType defines the type of container.
 type ContainerType string
 
+// BSRFileName defines the name of a bsr file.
+type BSRFileName string
+
+// BSRChecksumStatus defines the status of the checksum validation
+type BSRChecksumStatus string
```

Comment:
I think this should be a struct that has uses status enum value and an error field; if the status enum is FAILED, then GetFailedItems can get the error field for the specifics on why it failed

Reply:
I like this idea. 

@elimt Would you like GetFailedItems to return a map[string]string or would you prefer a map[string]struct, incase you need to handle any logic dealing with the status output.

Reply:
We wanted to support verbose mode and non-verbose mode. Using `GetFailedItems` to get the error field/specifics will work for the non-verbose mode.

For the verbose mode, if we want to return the status of every file do we expect the status to be strings like "failed" or will we return the status enum?

Reply:
I don't think we want to return the status enum because that will only look like a integer to the user, unless we add a String() method to the enum. Maybe we need a proto definition to share between the validation functions.

Reply:
I think the distinction between verbose and non-verbose mode is something for the cli layer to consider, and the bsr package should just return a structure that contains the full results in a way that is easily used programmatically (see my [comment here](https://github.com/hashicorp/boundary/pull/3504#discussion_r1275267614)). So the functionality of `GetFailedItems` to select just the files that failed, along with the logic of combining the status with the error would be something done in the cli.